ocaml/xapi/xapi_pool.ml

```diff
@@ -3487,36 +3511,27 @@ let get_updates_handler (req : Http.Request.t) s _ =
             ~self:(Helpers.get_pool ~__context)
             ~doc:"pool.get_updates" ~op:`get_updates (fun () ->
               try
-                let json_str =
-                  Yojson.Basic.to_string
-                    (Repository.get_pool_updates_in_json ~__context ~hosts)
-                in
-                let size = Int64.of_int (String.length json_str) in
-                Http_svr.headers s
-                  (Http.http_200_ok_with_content size ~keep_alive:false ()
-                  @ [Http.Hdr.content_type ^ ": application/json"]
-                  ) ;
-                Unixext.really_write_string s json_str |> ignore
+                http_res (Repository.get_pool_updates_in_json ~__context ~hosts)
               with
               | Api_errors.(Server_error (failure, _)) as e
                 when List.mem failure failures_of_404 ->
                   error "404: can't get updates for pool: %s"
                     (ExnHelper.string_of_exn e) ;
-                  Http_svr.headers s (Http.http_404_missing ())
+                  http_err failure []
               | Api_errors.(Server_error (failure, _)) as e
                 when not (List.mem failure failures_of_404) ->
                   error "getting updates for pool failed: %s"
                     (ExnHelper.string_of_exn e) ;
-                  raise e
+                  http_err failure []
```

Comment:
This is changing the behaviour of the code, isn't? Previously we had an exception whereas now we are returning normal responses. I wonder if this is ok to clients of xapi other than xencentre. Would it be better to preserve this raising of Api_error?